os/arch/xtensa/include/irq.h

```diff
@@ -0,0 +1,325 @@
+#ifdef CONFIG_ARCH_FAMILY_LX6
+#include <arch/lx6/irq.h>
+
+/* Include implementation-specific IRQ definitions (including IRQ numbers) */
+
+#ifdef CONFIG_ARCH_CHIP_ESP32
+#include <arch/esp32/irq.h>
+#else
+#error "Unknown LX6 implementation"
+#endif
+
+#else
+#error "Unknown XTENSA architecture"
+#endif
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* IRQ Stack Frame Format.  Each value is a uint32_t register index */
+
+#define REG_PC              (0)	/* Return PC */
+#define REG_PS              (1)	/* Return PS */
+#define REG_A0              (2)
+#define REG_A1              (3)	/* Stack pointer before interrupt */
+#define REG_A2              (4)
+#define REG_A3              (5)
+#define REG_A4              (6)
+#define REG_A5              (7)
+#define REG_A6              (8)
+#define REG_A7              (9)
+#define REG_A8              (10)
+#define REG_A9              (11)
+#define REG_A10             (12)
+#define REG_A11             (13)
+#define REG_A12             (14)
+#define REG_A13             (15)
+#define REG_A14             (16)
+#define REG_A15             (17)
+#define REG_SAR             (18)
+#define REG_EXCCAUSE        (19)
+#define REG_EXCVADDR        (20)
+
+#define _REG_LOOPS_START    (21)
+
+#ifdef XCHAL_HAVE_LOOPS
+#define REG_LBEG          (_REG_LOOPS_START + 0)
+#define REG_LEND          (_REG_LOOPS_START + 1)
+#define REG_LCOUNT        (_REG_LOOPS_START + 2)
+#define _REG_WINDOW_TMPS  (_REG_LOOPS_START + 3)
+#else
+#define _REG_WINDOW_TMPS  _REG_LOOPS_START
+#endif
+
+#ifndef __XTENSA_CALL0_ABI__
+  /* Temporary space for saving stuff during window spill.
+   * REVISIT: I don't think that we need so many temporaries.
+   */
+
+#define REG_TMP0          (_REG_WINDOW_TMPS + 0)
+#define REG_TMP1          (_REG_WINDOW_TMPS + 1)
+#define _REG_OVLY_START   (_REG_WINDOW_TMPS + 2)
+#else
+#define _REG_OVLY_START   _REG_WINDOW_TMPS
+#endif
+
+#ifdef CONFIG_XTENSA_USE_OVLY
+/* Storage for overlay state */
+
+#error Overlays not supported
+#define XCPTCONTEXT_REGS  _REG_OVLY_START
+#else
+#define XCPTCONTEXT_REGS  _REG_OVLY_START
+#endif
+
+#define XCPTCONTEXT_SIZE    (4 * XCPTCONTEXT_REGS)
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+#ifndef __ASSEMBLY__
+
+/* This struct defines the way the registers are stored. */
+
+struct xcptcontext {
+	/* The following function pointer is non-zero if there are pending signals
+	 * to be processed.
+	 */
+
+#ifndef CONFIG_DISABLE_SIGNALS
+	void *sigdeliver;			/* Actual type is sig_deliver_t */
+
+	/* These are saved copies of registers used during signal processing. */
+
+	uint32_t saved_pc;
+	uint32_t saved_ps;
+#endif
+
+	/* Register save area */
+
+	uint32_t regs[XCPTCONTEXT_REGS];
+
+#if XCHAL_CP_NUM > 0
```

Comment:
Where is `XCHAL_CP_NUM` definition?

Reply:
it's defined in #include <arch/chip/tie.h>
And the soc chip code will be added soon.

Reply:
XCHAL_CP_NUM is defined in the SOC chip header, it will be added soon.